contract/models/contract_line.py

```diff
@@ -818,6 +818,17 @@ def stop(self, date_end, manual_renew_needed=False, post_message=True):
                     }
                     if rec.last_date_invoiced == date_end:
```

Comment:
Do we have a test for this case (`rec.last_date_invoiced == date_end`)? Is it still a special case (at first glance I'd say it should be covered by `get_next_invoice_date`)?

Reply:
yes, 

https://github.com/OCA/contract/blob/659b91b0f0c50d29cf0b685fe9ab6bd1fb840543/contract/tests/test_contract.py#L2278-L2282

Reply:
@sbejaoui I still think the `== date_end` is now useless as this cased should be covered by the `get_next_invoice_date` in the else.

Reply:
I will test it

Reply:
True, get_next_invoice_date cover the case, I will make the change